src/server/src/main/java/io/cassandrareaper/service/RepairRunService.java

```diff
@@ -185,7 +196,45 @@ public RepairRun registerRepairRun(
       LOG.error(errMsg);
       throw new ReaperException(errMsg);
     }
-    return segments;
+
+    // Compute replicas per DC for each segment
+    List<Segment> segmentsWithReplicas = Lists.newArrayList();
+    for (Segment segment:segments) {
+      segmentsWithReplicas.add(
+          Segment.builder()
+                 .withBaseRange(segment.getBaseRange())
+                 .withTokenRanges(segment.getTokenRanges())
+                 .withReplicas(getDCsByNodeForRepairSegment(targetCluster, segment, repairUnit.getKeyspaceName()))
+                 .build());
+    }
+
+    return segmentsWithReplicas;
+  }
+
+  private Map<String, String> getDCsByNodeForRepairSegment(
+      Cluster cluster,
+      Segment segment,
+      String keyspace) throws ReaperException {
+
+    final int maxAttempts = 5;
```

Comment:
why `5`? In what time-frame can the 5 subsequent calls fail within?

Reply:
It's totally arbitrary. What's your take on this?

Reply:
IMHO it shouldn't be a hardcoded (nor magic) constant, and it should have a sleep in the loop.

Looking at the code I would think that when a node is down it is going to fail 5 times in a row reasonably quickly.

How long can/should this loop run for? How long should we be waiting for a node to become available again? What is this relative to, e.g. other settings in reaper?

Reply:
The idea here is rather to handle transient networking issues rather than handling nodes going down.
The way we implement this is that we'll go through `ClusterFacade.connect()` which will pick the fastest node answering in the cluster and then invoke `getRangeToEndpointMap()` on that node.
If we manage connecting to the node, and then it goes down before `getRangeToEndpointMap()`, then the loop will trigger the operation again, picking another node and trying to get the info from it. We won't end up trying to get the information from the same node throughout the retries.
That should be fairly resilient already I think and failing to create the repair run is ok as well (it'll get rescheduled by Reaper or the user will get some feedback and click the button again), so it's probably not worth waiting for too long there.

I can move the max number of attempts to a constant at the top, but I don't think we should make it configurable (probably no need to complexify the config file for this).
Looking at my above description of how things go, I think we could just perform a single retry to cover edge cases and fail fast instead of retrying too much.
wdyt?